vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java

```diff
@@ -1011,7 +1041,7 @@ private boolean processWays(QueryParameters queryParameters, ITileDataSink mapDa
                         for (int i = 0; i < e.tags.size(); i++)
                             tags.add(e.tags.get(i));
                         if (Selector.ALL == selector || hasName || hasHouseNr || hasRef || wayAsLabelTagFilter(tags)) {
-                            GeoPoint labelPos = e.labelPosition != null ? new GeoPoint(e.labelPosition.y / 1E6, e.labelPosition.x / 1E6) : null;
+                            GeoPoint labelPos = e.labelPosition != null ? new GeoPoint(labelPosition[0] / 1E6, labelPosition[1] / 1E6) : null;
```

Comment:
A null check for `labelPosition` array should be done here instead of `e.labelPosition`.

Reply:
correct, i missed that.

Reply:
Actually the old code should remain?
```java
GeoPoint labelPos = e.labelPosition != null ? new GeoPoint(e.labelPosition.y / 1E6, e.labelPosition.x / 1E6) : null;
```
(`Way.labelPosition` is not used anywhere anyway)

Reply:
the old code was not working at all. because at the time calling, e.labelPosition stores tileCoordinates. (it is called after project(e))
Dividing it by 1E6 will give, uhmn.. micro-tilecoordinates. don't know. Nothing like latlong for sure.

This should be changed. Maybe useful when a click on label needs to be detected.